Toolbar component options

JoomFX opened the pull request:
**Do not merge until the updated rendering is addressed in all suites**

closes https://github.com/telerik/kendo-themes/issues/3294

* [x] Add HTML helper
* [x] Use HTML helper in visual tests
* [x] Add sizing visual tests
* [x] Improve overflow popup button styles (remove `position: absolute;`)
* [x] Implement component options.
* [x] Variables (add prefix `$kendo-` if missing)
* [x] Add variables description
* [x] Create JSON structure
* [x] Docs article containing each mode.

Actions required
- [ ] Add sizing options and update rendering in jQuery
- [ ] Add sizing options and update rendering in MVC & Core
- [ ] Add sizing options and update rendering in Angular - https://github.com/telerik/k2/issues/676
- [ ] Add sizing options and update rendering in React - https://github.com/telerik/kendo-react-private/pull/2019
- [ ] Add sizing options and update rendering in Blazor - https://github.com/telerik/blazor/pull/5518
- [ ] Add sizing options and update rendering in Vue

epetrow commented:
@JoomFX I noticed a few discrepancies with the design related to the paddings and spacing between items.

Default, Bootstrap, Material:
- small: gap between items by design is 6px

Fluent:
- small: padding by design is 6px
- gap between items for all sizes by design is 8px

JoomFX commented:
I just had a discussion with @dmanova and here's the conclusion:

1. **Default, Bootstrap, Material** - the gap between the items in the small toolbar should indeed be 6px (larger than the toolbar padding). I fixed it already.
2. **Fluent** - the padding of the small toolbar should be 4px, it is now fixed in the Figma file. The gap between the items in all toolbar sizes should indeed be 8px (the same gap for all sizes). I fixed it already.

TeyaVes commented on packages/default/scss/grid/_variables.scss:
```diff
@@ -20,9 +20,9 @@ $grid-grouping-header-padding-y: $grid-grouping-header-padding-x !default;
-$grid-toolbar-padding-x: $toolbar-padding-x !default;
-$grid-toolbar-padding-y: $toolbar-padding-y !default;
-$grid-toolbar-spacing: $toolbar-spacing !default;
+$grid-toolbar-padding-x: $kendo-toolbar-padding-x-md !default;
```
the grid-toolbar-padding and spacing vars are not being used anywhere and can be deleted.

Juveniel commented:
@JoomFX please rebase the PR with develop and group commits (merge tests in single commit, merge styles and variable docs in single commit, move regen docs commit as last).

In addition to the above, please add respective breaking changes labels to the respective commits (for the rendering change and for the prefixed variables).

joneff reviewed (approved):
lgtm
